libminifi/include/sitetosite/RawSocketProtocol.h

```diff
@@ -143,21 +140,21 @@ class RawSiteToSiteClient : public sitetosite::SiteToSiteClient {
   // tearDown
   virtual void tearDown();
   // write Request Type
-  virtual int writeRequestType(RequestType type);
+  virtual int writeRequestType(RequestType type) {
+    return SiteToSiteClient::writeRequestType(type);
```

Comment:
I think the code being in SiteToSiteClient is vestigial. It probably shouldn't be there. That also would reduce the risk of any testing. Just remove it from the base class. The only other implementation ( and others I have in mind ) wouldn't need this.

Reply:
Okay, good shout, this helped to remove even more duplications. :]